generate.py

```diff
@@ -73,7 +73,7 @@ def generate(
             xm.mark_step()
 
         # concatenate the new generation
-        idx[t] = idx_next
+        idx[t] = idx_next.item()
```

Comment:
If this issue is MPS only, we should do this to avoid the performance hit to other backends

```suggestion
        # https://github.com/pytorch/pytorch/issues/101936
        idx[t] = idx_next.item() if idx.device.type == "mps" else idx_next
```